### DISCUSSION TITLE
How to show diff of failures?

### DISCUSSION THREAD

**Discussion by gaborbernat** [assertj/assertj#3644] [Q&A]
I am using junit 5 with assertj, but when a failure happens (for a 100 line string comparision) all I get is:

```
 org.opentest4j.AssertionFailedError:
    expected:
....

	but was:
....

        at app//main.BaseTester.assertPatchedCallstackOnDisk(BaseTester.groovy)
```

Is there a way to get something like git diff does?

**Comment by joel-costigliola** [comment]
Not really something AssertJ can control since what you get comes from opentest4j and the capabilities of your IDE.

The IDE could inspect the AssertionFailedError, which has the actual and expected values and if these are strings show a better diff

**Reply by gaborbernat** [reply]
What about in the CI outside of the IDE (using gradle..)?

**Reply by joel-costigliola** [reply]
I don't think they could do much.

Having said that, we could add an option to generate an error message with diff style syntax when actual and expected are strings but that would not be the default IMO.

**Reply by joel-costigliola** [reply]
Usually IDEs have visual diff showing actual and expected string representation in a diff editor

**Reply by gaborbernat** [reply]
Is there a Gradle plugin to take the AssertionFailedError and calc the diff? A flag would be also nice.

**Reply by joel-costigliola** [reply]
I'm not aware of any
